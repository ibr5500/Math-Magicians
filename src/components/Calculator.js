import React, { useState }from 'react';
import calculate from './logic/calculate';


const Calculator = () => {
    const [anyName, setState] = useState({
        total: 0,
        next: null,
        operation: null,
    });

    const handleClick = (event) => {
        setState(
            calculate(anyName, event.target.id)
        );
    };

    const {total, next, operation} = anyName;
    
    return (
        <section className='parent-container'>
            <h2>Let's do some math!</h2>
            <div className = "contianer">
                <div className='res'>
                    {total} {operation} {next}
                </div>
                <div className='row'>
                    <button type="button" id='AC' onClick={handleClick}>AC</button>
                    <button type="button" id='+/-' onClick={handleClick}>+/-</button>
                    <button type="button" id='%' onClick={handleClick}>%</button>
                    <button type="button" id='รท' className='ops' onClick={handleClick}>รท</button>
                </div>
                <div className='row'>
                    <button type="button" id='7' onClick={handleClick}>7</button>
                    <button type="button" id='8' onClick={handleClick}>8</button>
                    <button type="button" id='9' onClick={handleClick}>9</button>
                    <button type="button" id='x' className='ops' onClick={handleClick}>x</button>
                </div>
                <div className='row'>
                    <button type="button" id='4' onClick={handleClick}>4</button>
                    <button type="button" id='5' onClick={handleClick}>5</button>
                    <button type="button" id='6' onClick={handleClick}>6</button>
                    <button type="button" id='-' className='ops' onClick={handleClick}>-</button>
                </div>
                <div className='row'>
                    <button type="button" id='1' onClick={handleClick}>1</button>
                    <button type="button" id='2' onClick={handleClick}>2</button>
                    <button type="button" id='3' onClick={handleClick}>3</button>
                    <button type="button" id='+' className='ops' onClick={handleClick}>+</button>
                </div>
                <div className='row'>
                    <button type="button" id='0' className='zero' onClick={handleClick}>0</button>
                    <button type="button" id='.' onClick={handleClick}>.</button>
                    <button type="button" id='=' className='ops' onClick={handleClick}>=</button>
                </div>
            </div>
        </section>
    );
};

export default Calculator;