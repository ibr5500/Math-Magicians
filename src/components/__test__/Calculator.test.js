import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Test for the Calculator component', () => {
    it('Should render the Calculator', () => {
      const tree = renderer.create(<Calculator />);
      expect(tree).toMatchSnapshot();
    });
});