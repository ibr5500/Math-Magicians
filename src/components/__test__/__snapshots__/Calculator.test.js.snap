// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Calculator component Should render the Calculator 1`] = `
<section
  className="parent-container"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="contianer"
  >
    <div
      className="res"
    >
      0
       
       
    </div>
    <div
      className="row"
    >
      <button
        id="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        id="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        id="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="ops"
        id="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row"
    >
      <button
        id="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        id="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        id="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="ops"
        id="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row"
    >
      <button
        id="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        id="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        id="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="ops"
        id="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row"
    >
      <button
        id="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        id="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        id="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="ops"
        id="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="zero"
        id="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        id="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="ops"
        id="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
